lib/coverband/reporters/html_report.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Coverband
+  module Reporters
+    class HTMLReport < Base
+      def self.report(store, options = {})
+        scov_style_report = super(store, options)
+        open_report = options.fetch(:open_report) { true }
+        html = options.fetch(:html) { false }
+
+        # list all files, even if not tracked by Coverband (0% coverage)
+        tracked_glob = "#{current_root}/{app,lib,config}/**/*.{rb}"
+        report_files = Coverband::Utils::Result.add_not_loaded_files(scov_style_report, tracked_glob)
+        # apply coverband filters
+        filtered_report_files = {}
+        report_files.each_pair do |file, data|
+          next if Coverband.configuration.ignore.any? { |i| file.match(i) }
+          filtered_report_files[file] = data
+        end
+
+        if html
+          Coverband::Utils::HTMLFormatter.new(filtered_report_files).format_html!
+        else
+          Coverband::Utils::HTMLFormatter.new(filtered_report_files).format!
+          if open_report
+            `open #{Coverband.configuration.root}/coverage/index.html`
+          else
+            Coverband.configuration.logger.info 'report is ready and viewable: open coverage/index.html'
+          end
+
+          Coverband::Utils::S3Report.instance.persist! if Coverband.configuration.s3_bucket
```

Comment:
Yeah that was a question I had as well... Is there any good use cases for generating the reporting and writing to S3... 

What I came up with:
* If you didn't want to use the Rack-mountable endpoints at all
* Archiving... if you write to S3 with the audit function of S3 it will have all your coverage history (at least every version you uploaded)
* CI based workflows, where CI is just generating and making reports like this visible (what I used to do before the rack mountable endpoint)

I can't decide how many folks want these types of features vs mounting the new web piece.

Reply:
All good points. Difficult to get a handle on what features people need. Almost need coverband for coverband 🤔 .

How about we deprecate the s3 report for this release? Could put a log message and a note in the README.